### ISSUE TITLE
`no-unused-var` assignment operator edge case

### ISSUE THREAD

**Issue by emilbayes** [eslint/eslint#1166]
The `no-unused-var` rule does not catch this edge case where an assignment operator is used to update a variable, and the updated value is used directly inside a function call (or any other place where an expression is allowed):

``` js
function accelerate(fn) {
    var currentSpeed = 1; //Error
    return function(multiplier) {
        fn(currentSpeed *= multiplier); //Assignment
    };
}
```

I'm using the master branch of ESLint. I do not have time to look into this myself at the moment. I hope I've used the right terminology :)
